### PR TITLE
fix task in recovered balance plan is still marked as FAILED

### PR DESCRIPTION
**Pull request by critical27**
When task is failed for some reason, when recovering, they are marked as `IN_PROGRESS` again, but `balance data id` will still show `FAILED`. Just save the status when plan is recovered.

### CODE REVIEW

**Review by panda-sheep — DISMISSED** [review]
Well done！

**Comment by critical27** [comment]
There will be more changes, hold on one second.

**Comment by critical27** [comment]
close https://github.com/vesoft-inc/nebula-storage/issues/458
close https://github.com/vesoft-inc/nebula-storage/issues/458

File: src/meta/processors/admin/AdminClient.cpp [hunk 1]
```diff
@@ -36,6 +36,10 @@ folly::Future<Status> AdminClient::transLeader(GraphSpaceID spaceId,
     if (it == peers.end()) {
         return Status::PartNotFound();
     }
+    if (peers.size() == 1 && peers.front() == leader) {
```
Comment (darionyaphet) [inline src/meta/processors/admin/AdminClient.cpp:39]: make sure the leader is being selected?

Comment (critical27) [reply src/meta/processors/admin/AdminClient.cpp:39]: No, transfer leader will failed in some case when replica is 1. see https://github.com/vesoft-inc/nebula-storage/issues/458.

**Review by panda-sheep — APPROVED** [review]

**Review by liuyu85cn — APPROVED** [review]
LGTM

**Pull request merged**